### ISSUE TITLE
Sending a content-type with a charset specified in requests to docker prevent correct interactions with authorization plugins present on the daemon.

### ISSUE THREAD

**Issue by nathanburrell** [fabric8io/docker-maven-plugin#945]
Due to a [bug 36317](https://github.com/moby/moby/issues/36317) with docker in how it chooses whether or not to forward bodies of requests to authorization plugins, the maven plugin cannot make any successful requests if the docker daemon uses the body in any authorization plugins present.

Docker will most likely fix this in a newer version, however as this plugin seems to be compatible with older docker daemons, line 120 of [ApacheHttpClientDelegate](https://github.com/fabric8io/docker-maven-plugin/blob/master/src/main/java/io/fabric8/maven/docker/access/hc/ApacheHttpClientDelegate.java) could be updated to just pass the content type **" application/json"** with no charset to allow the plugin to interact with older daemons and authorization plugins correctly.

Steps to reproduce:

1. Install the debug auth plugin I created to demonstrate this (it simply logs the content length, content type and the length of the body).

https://bitbucket.org/nburrell/authz/src/master/

2. Use the attached pom file (changing the extension to xml due to github not allowing uploads of xml files) and run mvn clean  to docker (note the content-type is simply "application/json") and note the length of the body being the same as the content length in the request:

[pom.txt](https://github.com/fabric8io/docker-maven-plugin/files/1726583/pom.txt)

**Comment by rhuss** [comment]
@nathanburrell makes sense to me. Fancy doing a pull request ? (sorry, I'm on the road this week, but will fix this when being back).

BTW, dmp has an API detection mechanism and react differently on different Docker versions. Not sure whether this is usable at that level though.

**Comment by StephenFox1995** [comment]
I've made a PR to fix this issue https://github.com/fabric8io/docker-maven-plugin/pull/956
